dbt_schema_builder/schema.py

```diff
@@ -1,137 +1,69 @@
+            raise InvalidConfigurationException(
+                'Schema {} has both INCLUDE and EXCLUDE sections in its'
+                'sections in its configuration file'.format(schema_name)
             )
-        elif current_safe_source:
-            for source in self.new_downstream_sources["sources"]:
-                if source["name"] == self.safe_downstream_source_name:
-                    source["tables"].append(current_safe_source)
-                elif source["name"] == self.pii_downstream_source_name:
-                    source["tables"].append(current_pii_source)
-        else:
-            for source in self.new_downstream_sources["sources"]:
-                if source["name"] == self.safe_downstream_source_name:
-                    source["tables"].append(
-                        {
-                            "name": relation.relation,
-                            "description": DEFAULT_DESCRIPTION,
-                        }
-                    )
-                elif source["name"] == self.pii_downstream_source_name:
-                    source["tables"].append(
-                        {
-                            "name": relation.relation,
-                            "description": DEFAULT_DESCRIPTION,
-                        }
-                    )
 
-    def update_trifecta_models(self, relation):
-        """
-        Given a relation, add it to the 'trifecta'. These are the PII and safe views
-        constructed from the raw data.
-        """
-        for relation_name in [
-            relation.new_pii_relation_name,
-            relation.new_safe_relation_name,
-        ]:
-            self.add_model_to_new_schema(
-                relation_name, relation.meta_data
-            )
+        schema = Schema(
+            schema_name,
+            exclusion_list,
+            inclusion_list,
+            relations=[]
+        )
+        return schema
 
-    def add_model_to_new_schema(self, new_relation_name, model_meta_data):
+    def filter_relations(self):
         """
-        Add models and their columns to a schema that is currently being generated.
+        Filter the relations in this Schema, based upon the exclusion and
+        inclusion lists.
         """
-        # Add our table to the "models" list in the new schema
-        self.new_schema["models"].append({"name": new_relation_name})
+        filtered_relations = []
+        for relation in self.relations:
+
+            if self.exclusion_list and not self.inclusion_list:
+                if relation.source_relation_name not in self.exclusion_list:
+                    filtered_relations.append(relation)
+            elif not self.exclusion_list and self.inclusion_list:
+                if relation.source_relation_name in self.inclusion_list:
+                    filtered_relations.append(relation)
+            elif not self.exclusion_list and not self.inclusion_list:
+                filtered_relations.append(relation)
```

Comment:
I'd feel slightly better if we had the `else` here to throw an error even though we are checking upstream, just to make sure it never hits this condition.

Reply:
Is it possible that we may get in this state? There is only one fast moving table identified for Disco. What about new IDAs?

Reply:
I think that since it's an error upstream it should just be an error here if something is in both the include and exclude list.

Reply:
I don't think this is actually an error. We have 2 raw schemas that are currently operated on by schema builder that do not need the stitch/sqoop hybrid (no fast moving tables detected).

Reply:
So in that case it would hit the `not self.exclusion_list and not self.inclusion_list` right? What is the case for having both an exclusion list and an inclusion list? Right now this code would just silently skip all relations for that raw schema, wouldn't it?